provider.go

```diff
@@ -306,3 +328,22 @@ func (wp *WasmcloudProvider) ToActor(actorID string, msg []byte, op string) ([]b
 
 	return resp.Msg, nil
 }
+
+func (wp *WasmcloudProvider) putLink(l core.LinkDefinition) {
+	wp.lock.Lock()
```

Comment:
Nit: Would it technically be a bit safer in these new functions to just do a `defer wp.lock.Unlock()`? The lock in all these cases should last for the life of the function. If they were more complex and we wanted to unlock earlier it would make more sense to manually unlock

Reply:
Yes, since these functions are simple, using defer is better. Done.